asynch/pool.py

```diff
@@ -140,26 +140,46 @@ async def _create_connection(self) -> None:
         if pool_size > maxsize:
             raise AsynchPoolError(f"{self} is overburden")
 
-        conn = await connect(**self._connection_kwargs)
+        conn = Connection(**self._connection_kwargs)
+        await conn.connect()
         self._free_connections.append(conn)
 
-    async def _acquire_connection(self) -> Connection:
+    def _pop_connection(self) -> Connection:
         if not self._free_connections:
             raise AsynchPoolError(f"no free connection in {self}")
 
-        conn = self._free_connections.popleft()
+        return self._free_connections.popleft()
+
+    async def _acquire_connection(self) -> Connection:
+        conn = self._pop_connection()
+
+        try:
+            await conn._refresh()
+        except ConnectionError:
+            await self._create_connection()
+            conn = self._pop_connection()
```

Comment:
Checking if a connection candidate is fresh (alive) and if not, creating the new one to substitute the invalid one.